### DISCUSSION TITLE
Adding store slices may cause naming conflicts

### DISCUSSION THREAD

**Discussion by yibird** [pmndrs/zustand#1538] [Q&A]
Examples of document slicing are as follows:

```ts
import {createAppSlice,createTabSlice} from './slices';

type RootState = AppSlice & TabSlice;
export const useStore = create<RootState>()((...args) => ({
  ...createAppSlice(...args),
  ...createTabSlice(...args),
}));
```

expect:
```ts
import {createAppSlice,createTabSlice} from './slices';

type RootState ={
  app:AppSlice,
  tab:TabSlice
};
export const useStore = create<RootState>()((...args) => ({
  app:createAppSlice(...args),
  tab:createTabSlice(...args),
}));
````
But the args type is wrong

**Comment by dai-shi** [comment]
That's what we call namespacing slices, which is more advanced.
Some third party libraries support namespacing. For example: zustand-lens in https://github.com/pmndrs/zustand/blob/3c03e40f93e20eb28ab50fc84a7e64af92bf63c5/docs/integrations/third-party-libraries.md

**Reply by Sharonzd** [reply]
Does Zustand recommend an official pattern to avoid the state name conflicts of different slices?

**Reply by dai-shi** [reply]
Zustand library doesn't handle namespacing (more precisely slicing too), so it's a field for third party libs.
For some patterns, we welcome improving docs. So far, I don't think we have a pattern to avoid name conflicts.

**Reply by Sharonzd** [reply]
Got it, thanks for your response.
